lib/remote.provider.jstp.api.js

```diff
@@ -297,7 +298,38 @@ const createRemoteProviderJstpApi = (gsProvider, cursorFactory) => ({
     },
 
     getSchemaSources(connection, callback) {
-      gsProvider.getSchemaSources(callback);
+      const userId = connection.session.get('userId');
+      const ops = [
+        (ctx, callback) => {
+          if (!userId) {
+            gsProvider.listActions((err, actions) => {
+              ctx.actionLists = { public: actions.public, private: {} };
+              callback(err);
+            });
+          } else {
+            gsProvider.listActions((err, actions) => {
+              ctx.actionLists = actions;
+              callback(err);
+            }, createEntityFilterer(gsProvider, userId, filterActions));
+          }
+        },
+      ];
+      if (userId) {
+        ops.push((ctx, callback) => {
+          gsProvider.listCategories((err, categories) => {
+            ctx.categoryList = categories;
+            callback(err);
+          }, createEntityFilterer(gsProvider, userId, filterCategories));
+        });
+      }
```

Comment:
Nit: I think it 'll be clearer if we split it this way:
```js
const ops = [];
if (!userId) {
  ops.push((ctx, callback) => {
    gsProvider.listActions((err, actions) => {
      ctx.actionLists = { public: actions.public, private: {} };
      callback(err);
    });
  }));
} else {
  ops.push(
    (ctx, callback) => {
      gsProvider.listActions((err, actions) => {
        ctx.actionLists = actions;
        callback(err);
      }, createEntityFilterer(gsProvider, userId, filterActions));
    },
    (ctx, callback) => {
      gsProvider.listCategories((err, categories) => {
        ctx.categoryList = categories;
        callback(err);
      }, createEntityFilterer(gsProvider, userId, filterCategories));
    }
  );
}
```
This also avoids double checking `userId`.